lib/experimental/fonts-library/class-wp-rest-fonts-library-controller.php

```diff
@@ -0,0 +1,215 @@
+
+		return new WP_REST_Response( __( 'Font family uninstalled successfully.', 'gutenberg' ), 200 );
+	}
+
+	/**
+	 * Check if user has permissions to update the fonts library.
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_fonts_library_permissions_check() {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_update_fonts_library',
+				__( 'Sorry, you are not allowed to update the fonts library on this site.', 'gutenberg' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		// The update endpoints requires write access to the temp and the fonts directories.
+		$temp_dir = get_temp_dir();
+		if ( ! is_writable( $temp_dir ) || ! wp_is_writable( WP_FONTS_DIR ) ) {
+			return new WP_Error(
+				'rest_cannot_write_fonts_folder',
+				__( 'Error: WordPress does not have permission to write the fonts folder on your server.', 'gutenberg' ),
+				array(
+					'status' => 500,
+				)
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if user has permissions to read the fonts library.
+	 *
+	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function read_fonts_library_permissions_check() {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_cannot_read_fonts_library',
+				__( 'Sorry, you are not allowed to read the fonts library on this site.', 'gutenberg' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Fetches the Google Fonts JSON file.
+	 *
+	 * Reads the "google-fonts.json" file from the file system and returns its content.
+	 *
+	 * @return WP_REST_Response|WP_Error The content of the "google-fonts.json" file wrapped in a WP_REST_Response object.
+	 */
+	public function get_google_fonts() {
+		$file = file_get_contents(
+			path_join( dirname( __FILE__ ), 'google-fonts.json' )
+		);
+		if ( $file ) {
+			return new WP_REST_Response( json_decode( $file ) );
+		}
+		return new WP_Error(
+			'rest_cant_read_google_fonts',
+			__( 'Error reading Google Fonts JSON file.', 'gutenberg' ),
+			array( 'status' => 500 )
+		);
+	}
+
+	/**
+	 * Installs new fonts.
+	 *
+	 * Takes a request containing new fonts to install, downloads their assets, and adds them to the fonts library.
+	 *
+	 * @param WP_REST_Request $request The request object containing the new fonts to install in the request parameters.
+	 * @return WP_REST_Response|WP_Error The updated fonts library post content.
+	 */
+	public function install_fonts( $request ) {
+		// Get new fonts to install.
+		$fonts_to_install = $request->get_param( 'fontFamilies' );
+
+		// As we are receiving form data, the font families are encoded as a string.
+		// We are using form data because local fonts need to use that format to attach the files in the request.
```

Comment:
```suggestion
		/*
		 * As we are receiving form data, the font families are encoded as a string.
		 * We are using form data because local fonts need to use that format to attach the files in the request.
		 */
```